### PR TITLE
Add support for HTTP 1.1 in StowRS requests

### PR DESCRIPTION
**Pull request by dzelemba**
Brings back old version of StowRS that was removed when we upgraded to HTTP 2.0 to improve error handling and sets it as the default.

This fixes flakiness issues with HTTP 2.0 in the Healthcare API.

Tested by running with the flag on/off and confirmed that the default has no errors whereas turning this flag on has about 20% error rate when running adapter locally on my laptop.

### CODE REVIEW

File: import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java [hunk 1]
```diff
@@ -89,10 +89,16 @@ public static void main(String[] args) throws IOException, GeneralSecurityExcept
...
-    IDicomWebClient defaultCstoreDicomWebClient =
+    IDicomWebClient defaultCstoreDicomWebClient = null;
+    if (flags.useHttp2ForStow) {
+      defaultCstoreDicomWebClient =
         new DicomWebClientJetty(
             credentials,
             StringUtil.joinPath(cstoreDicomwebAddr, cstoreDicomwebStowPath));
+    } else {
+      defaultCstoreDicomWebClient =
+        new DicomWebClient(requestFactory, flags.dicomwebAddress);
```
Comment (danielbeaudreau) [inline import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java:100]: Just using flags.dicomwebAddress isn't backwards compatible with the old way of specifying flags.dicomwebAddr + flags.dicomwebStowPath which we used to concatenated together and then make a request here.

To make this backwards compatible we'd need to create the address as 
StringUtil.joinPath(cstoreDicomwebAddr, cstoreDicomwebStowPath), and then not append "/studies" in the stow function.

Comment (dzelemba) [reply import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java]: Thanks for catching this. I think I have fixed it, but please take a close look.

I've filled https://github.com/GoogleCloudPlatform/healthcare-dicom-dicomweb-adapter/issues/85 to add an integration test for this in the future.

**Comment by dzelemba** [comment]
Fixed the issue and also added an integration test to cover the http2 functionality.

File: integration_test/README.md [hunk 2]
```diff
@@ -7,15 +7,14 @@ Creates a dicom store, builds & runs adapter, builds & runs store-scu against it
 Local run:
 
 ```shell
-cloud-build-local --config=integration_test/cloudbuild-integration-test.yaml --dryrun=false .
+cloud-build-local --config=integration_test/cloudbuild-integration-test.yaml --dryrun=false --substitutions=SHORT_SHA=local_run .
```
[the remark this answers is not in the document]
Comment (dzelemba) [reply integration_test/README.md:10]: Ya but when you run locally the SHORT_SHA value isn't bound to anything so it fails, so you have to just provide some value here.

File: export/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ExportAdapter.java [hunk 3]
```diff
@@ -58,8 +58,13 @@ public static void main(String[] args) throws IOException {
...
       credentials = credentials.createScoped(Arrays.asList(flags.oauthScopes.split(",")));
     }
+
+    boolean isLegacyAdress = flags.peerDicomwebAddress.isEmpty();
+    String peerDicomwebAddress =
+        isLegacyAdress ? flags.peerDicomwebAddr : flags.peerDicomwebAddress;
+    String peerDicomwebStowpath = isLegacyAdress ? flags.peerDicomwebStowPath : "studies";
     DicomWebClient dicomWebClient =
-        new DicomWebClient(createHttpRequestFactory(credentials), flags.dicomwebAddr);
+        new DicomWebClient(createHttpRequestFactory(credentials), peerDicomwebAddress, peerDicomwebStowpath);
```
[the remark this answers is not in the document]
Comment (dzelemba) [reply export/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ExportAdapter.java]: Got it. Ok went back to using flags.dicomwebAddr and used "studies" for now although if anyone tried to use the stow function here it wouldn't work.

File: import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java [hunk 4]
```diff
@@ -104,7 +110,7 @@ public static void main(String[] args) throws IOException, GeneralSecurityExcept
 
     // Handle C-FIND
     IDicomWebClient dicomWebClient =
-        new DicomWebClient(requestFactory, flags.dicomwebAddress);
+        new DicomWebClient(requestFactory, flags.dicomwebAddress, cstoreDicomwebStowPath);
```
[the remark this answers is not in the document]
Comment (dzelemba) [reply import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/ImportAdapter.java]: Done.

**Comment by dzelemba** [comment]
Made the updates you requested, but there seems to be a failure in the integration test that I can't reproduce locally so still looking into that.

**Comment by dzelemba** [comment]
Looks like the flakiness was just due to running multiple tests in parallel.

**Pull request merged**